### PR TITLE
Test exec async ordering

### PR DESCRIPTION
**Pull request by agerard-godaddy**
## Summary

Additional tests to document and verify expected behavior with `gasket.exec` lifecycle ordering.

### CODE REVIEW

File: packages/gasket-engine/test/exec.test.js [hunk 1]
```diff
@@ -81,4 +137,28 @@ describe('The exec method', () => {
 
     await expect(engine.exec('mock')).rejects.toThrow(Error);
   });
+
+  it('should await if ordered', async function () {
```
Comment (kbader-godaddy) [inline packages/gasket-engine/test/exec.test.js:141]:
```suggestion
  it('should await if ordered', async () => {
```
Small change to conform to how the rest of the tests are written.

File: packages/gasket-engine/test/exec.test.js [hunk 2]
```diff
@@ -81,4 +137,28 @@ describe('The exec method', () => {
...
+    expect(result).toEqual([
+      expect.objectContaining({ value: 'B' }),
+      expect.objectContaining({ value: 'A' })
+    ]);
+
+    const [resultsB, resultsA] = result;
+    expect(resultsA.start).toBeGreaterThanOrEqual(resultsB.end);
+  });
+
+  it('should await if ordered within first grouping', async function () {
```
Comment (kbader-godaddy) [inline packages/gasket-engine/test/exec.test.js:153]:
```suggestion
  it('should await if ordered within first grouping', async () => {
```
Same here.

File: packages/gasket-engine/test/exec.test.js [hunk 3]
```diff
@@ -10,6 +14,34 @@ describe('The exec method', () => {
     hooks: {
       eventA() {
         return Promise.resolve(1);
+      },
+      eventB: {
+        timing: {
+          after: ['pluginB']
+        },
+        handler: async function () {
```
Comment (kbader-godaddy) [inline packages/gasket-engine/test/exec.test.js:22]:
```suggestion
        async handler() {
```
Should we consider employing a named function in this context as a best practice to potentially enhance debugging by providing more informative stack traces?

File: packages/gasket-engine/test/exec.test.js [hunk 4]
```diff
@@ -10,6 +14,34 @@ describe('The exec method', () => {
...
+            start,
+            end: Date.now()
+          };
+        }
+      },
+      eventC: {
+        timing: {
+          first: true
+        },
+        handler: async function () {
```
Comment (kbader-godaddy) [inline packages/gasket-engine/test/exec.test.js:36]:
```suggestion
        async handler() {
```
Same here?

File: packages/gasket-engine/test/exec.test.js [hunk 5]
```diff
@@ -19,6 +51,30 @@ describe('The exec method', () => {
...
+          start,
+          end: Date.now()
+        };
+      },
+      eventC: {
+        timing: {
+          first: true,
+          before: ['pluginA']
+        },
+        handler: async function () {
```
Comment (kbader-godaddy) [inline packages/gasket-engine/test/exec.test.js:69]:
```suggestion
        async handler() {
```
Same here?

**Review by kbader-godaddy — APPROVED** [review]
Looks great! Couple small code quality questions.

**Review by jpage-godaddy — APPROVED** [review]

**Pull request merged**